src/modules/ekf2/EKF/terrain_control.cpp

```diff
@@ -106,21 +107,29 @@ void Ekf::updateTerrainValidity()
 		float hagl_var = INFINITY;
 		sym::ComputeHaglInnovVar(P, 0.f, &hagl_var);
 
-		if (hagl_var < fmaxf(sq(0.1f * getHagl()), 0.2f)) {
-			valid_hagl_var = true;
+		positive_hagl_var = hagl_var > 0.f;
+
+		if (positive_hagl_var
+		    && (hagl_var < sq(fmaxf(0.1f * getHagl(), 0.5f)))
```

Comment:
Any particular reason for 0.2->0.5? Should it be based on anything or configurable?

Reply:
It's just that we now square it and 0.5^2 ~= 0.2